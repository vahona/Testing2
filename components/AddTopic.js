import React, { useState } from "react";
export default function AddTopic({ addNewTopic }) {
  const [topicTitle, setTopicTitle] = useState("");
  return (
    <section>
      <header>
        <h4 data-testid="add-topic-title">Add a topic</h4>
      </header>
      <article>
        <form
          onSubmit={(e) => {
            e.preventDefault();
            e.target.reset();
            addNewTopic(topicTitle);
          }}
          className="add-topic"
        >
          <input
            data-testid="topic-title-input"
            required
            type="text"
            name="topic"
            placeholder="Type your topic idea here"
            value={topicTitle}
            onChange={(e) => setTopicTitle(e.target.value)}
          />
          <button type="submit" data-testid="submit-button">
            Submit
          </button>
        </form>
      </article>
    </section>
  );
}
