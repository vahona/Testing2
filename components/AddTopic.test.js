import React from "react";
import {
  render,
  fireEvent,
  getByTestId,
  cleanup,
} from "@testing-library/react";

import AddTopic from "./AddTopic";

afterEach(() => {
  cleanup();
  addNewTopic.mockClear();
});



// we create a spy/mock function
const addNewTopic = jest.fn();
test("<AddTopic/>", () => {
  const { debug, queryByText, getByText, queryByTestId, container } = render(
    <AddTopic />
  );
  // expect(queryByText('Add a topic')).toBeTruthy();
  expect(queryByTestId("add-topic-title")).toBeTruthy();
  expect(container.firstChild).toMatchSnapshot();
});


// i want to see if the function is called when I click on the submit button
test("<AddTopic addNewTopic={addNewTopic}/>", () => {
  const { getByTestId } = render(<AddTopic addNewTopic={addNewTopic} />);
  // first get the submit button
  expect(getByTestId("submit-button")).toBeTruthy();
  // add a topic to the text input
  // THIS WAY DOESN'T WORK
  // getByTestId('topic-title-input').value = 'hello onja';
  // fireEvent.change(getByTestId('topic-title-input'));
  // THIS WAY DOES
  fireEvent.change(getByTestId("topic-title-input"), {
    target: { value: "hello onja" },
  });

  
  // fire a click event on it
  fireEvent.click(getByTestId("submit-button"));
  expect(addNewTopic).toHaveBeenCalledWith("hello onja");
  // check if it was called with a new topic
});
