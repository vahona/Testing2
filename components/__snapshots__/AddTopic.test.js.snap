// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddTopic/> 1`] = `
<section>
  <header>
    <h4
      data-testid="add-topic-title"
    >
      Add a topic
    </h4>
  </header>
  <article>
    <form
      class="add-topic"
    >
      <input
        data-testid="topic-title-input"
        name="topic"
        placeholder="Type your topic idea here"
        required=""
        type="text"
        value=""
      />
      <button
        data-testid="submit-button"
        type="submit"
      >
        Submit
      </button>
    </form>
  </article>
</section>
`;
